Source/Scene/PickDepth.js

```diff
@@ -1,115 +1,36 @@
-    !defined(pickDepth._debugPickDepthViewportCommand) ||
-    useLogDepth !== pickDepth._useLogDepth
-  ) {
-    var fsSource =
-      "uniform highp sampler2D u_texture;\n" +
-      "varying vec2 v_textureCoordinates;\n" +
-      "void main()\n" +
-      "{\n" +
-      "    float z_window = czm_unpackDepth(texture2D(u_texture, v_textureCoordinates));\n" +
-      "    z_window = czm_reverseLogDepth(z_window); \n" +
-      "    float n_range = czm_depthRange.near;\n" +
-      "    float f_range = czm_depthRange.far;\n" +
-      "    float z_ndc = (2.0 * z_window - n_range - f_range) / (f_range - n_range);\n" +
-      "    float scale = pow(z_ndc * 0.5 + 0.5, 8.0);\n" +
-      "    gl_FragColor = vec4(mix(vec3(0.0), vec3(1.0), scale), 1.0);\n" +
-      "}\n";
-    var fs = new ShaderSource({
-      defines: [useLogDepth ? "LOG_DEPTH" : ""],
-      sources: [fsSource],
-    });
-
-    pickDepth._debugPickDepthViewportCommand = context.createViewportQuadCommand(
-      fs,
-      {
-        uniformMap: {
-          u_texture: function () {
-            return pickDepth._depthTexture;
-          },
-        },
-        owner: pickDepth,
-      }
-    );
-
-    pickDepth._useLogDepth = useLogDepth;
-  }
-
-  pickDepth._debugPickDepthViewportCommand.execute(context, passState);
 }
 
-function destroyTextures(pickDepth) {
-  pickDepth._depthTexture =
-    pickDepth._depthTexture &&
-    !pickDepth._depthTexture.isDestroyed() &&
-    pickDepth._depthTexture.destroy();
-}
-
-function destroyFramebuffers(pickDepth) {
-  pickDepth._framebuffer =
-    pickDepth._framebuffer &&
-    !pickDepth._framebuffer.isDestroyed() &&
-    pickDepth._framebuffer.destroy();
-}
-
-function createTextures(pickDepth, context, width, height) {
-  pickDepth._depthTexture = new Texture({
-    context: context,
-    width: width,
-    height: height,
-    pixelFormat: PixelFormat.RGBA,
-    pixelDatatype: PixelDatatype.UNSIGNED_BYTE,
-  });
-}
-
-function createFramebuffers(pickDepth, context, width, height) {
-  destroyTextures(pickDepth);
-  destroyFramebuffers(pickDepth);
-
-  createTextures(pickDepth, context, width, height);
-
-  pickDepth._framebuffer = new Framebuffer({
-    context: context,
-    colorTextures: [pickDepth._depthTexture],
-    destroyAttachments: false,
-  });
-}
+Object.defineProperties(PickDepth.prototype, {
+  framebuffer: {
+    get: function () {
+      return this._framebuffer.getFramebuffer();
+    },
+  },
+});
 
 function updateFramebuffers(pickDepth, context, depthTexture) {
   var width = depthTexture.width;
   var height = depthTexture.height;
 
-  var texture = pickDepth._depthTexture;
-  var textureChanged =
-    !defined(texture) || texture.width !== width || texture.height !== height;
-  if (!defined(pickDepth._framebuffer) || textureChanged) {
-    createFramebuffers(pickDepth, context, width, height);
+  var textureChanged = pickDepth._framebuffer.isDirty(width, height);
+  if (!defined(pickDepth.framebuffer) || textureChanged) {
```

Comment:
Is the `!defined(pickDepth.framebuffer)` part necessary?

Reply:
The logic would be different without it unless `isDirty` also checked that `framebuffer` was defined, but I don't think that's correct if the attachments are being set manually. It could be made to work either way, I'm not sure I have a preference.

Reply:
Possibly relevant discussion https://github.com/CesiumGS/cesium/pull/9966#discussion_r771014464:
> Maybe FramebufferManager should consider itself dirty if the framebuffer is undefined

Then the only code needed here would be `pickDepth._framebuffer.update(context, width, height);`